app/javascript/components/form/encounter/EncounterContactBarrier.vue

```diff
@@ -51,81 +52,60 @@ export default {
 				// the key has no mention of which actor
 				key = key.replace(new RegExp('_' + matcher), '');
 				// put the instrument and name in the arguments
-				transArgs.instrument = this.contactData[`${this.actor}InstrumentName`];
+				transArgs.instrument = this.state.instrumentName(this.actor);
 				transArgs.name = this.personName;
 				// pluralize as needed
 				transArgs.count = transArgs.instrument.match(/[^s]s$/) ? 1 : 0;
 			}
+
 			return this.$_t(key, transArgs);
 		},
-		inputValue: function() {
-			return this.barrier.key;
-		},
-		valInd: function() {
-			return this.modelValue.indexOf(this.inputValue);
-		},
-		cListeners: function() {
-			let vm = this;
-			return Object.assign({}, this.$listeners, {
-				change: function(e) {
-					vm.toggleChecked(e.target.checked);
-				}
-			});
-		},
-		actor: function() {
-			let matches = this.barrier.key.match(new RegExp(matcher));
-			return matches ? matches[0] : null;
-		},
-		canClean: function() {
+		canClean() {
+			// if it's not relevant whether it can be cleaned, return true
 			if (this.actor == null) {return true;}
-			let instID = this.contactData[this.actor + '_instrument_id'];
+			// get whether the current instrument this applies to is cleanable
+			let instID = this.state[this.actor + '_instrument_id'];
 			if (instID) {
-				return this.contactData.instruments[instID].can_clean;
+				return this.state.instruments[instID].can_clean;
 			}
+
+			// if there's no instrument, it can't be cleaned
 			return false;
 		},
-		shouldShow: function() {
+		shouldShow() {
 			// if it's a clean type, only continue if it's a cleanable instrument
 			if (this.barrier.key.includes('clean') && !this.canClean) {return false; }
 			// if it has no conditions, show it if it's cleanable
-			if (!this.barrier.encounter_conditions && !this.barrier.contact_conditions) { return this.canClean; }
+			if (!this.barrier.encounter_conditions
+				&& !this.barrier.contact_conditions) { return this.canClean; }
 
 			// if it has encounter conditions but all are false, don't show
-			if (this.barrier.encounter_conditions && this.encounterData && this.barrier.encounter_conditions.every((c) => !this.encounterData[c](this.contact))) { return false; }
+			if (this.barrier.encounter_conditions &&
+				(!this.tracker ||
+				this.barrier.encounter_conditions.every((c) => {
+					return !this.tracker[c](this.state.contact);
+				}))
+			) { return false; }
 
 			// if it has contact_conditions but some are false, don't show
-			if (this.barrier.contact_conditions && this.barrier.contact_conditions.some((c) => !this.contactData[c])) { return false; }
+			if (this.barrier.contact_conditions &&
+				this.barrier.contact_conditions.some((c) => {
+					return !this.state[c];
+				})
+			) { return false; }
```

Comment:
unfortunately no, because we get to these if statements if either type of condition exists, so they could still be undefined by the time we're searching them. I'm going to make `encounter_conditions` and `contact_conditions` computed variables that return an empty array instead of undefined to avoid having to do these checks

Reply:
ohhhhh. that's my bad. i thought you were checking `contact_conditions` for both `.every()` and `.some()`